crates/wasi-runtime-config/src/lib.rs

```diff
@@ -0,0 +1,68 @@
+use anyhow::Result;
+use std::collections::HashMap;
+
+mod gen_ {
+    wasmtime::component::bindgen!({
+        path: "wit",
+        world: "wasi:config/imports",
+        trappable_imports: true,
+    });
+}
+use self::gen_::wasi::config::runtime as generated;
+
+/// Capture the state necessary for use in the `wasi-runtime-config` API implementation.
+pub struct WasiRuntimeConfigCtx {
+    runtime_config: HashMap<String, String>,
+}
+
+impl WasiRuntimeConfigCtx {
+    /// Create a new context.
+    pub fn new<S, I>(config: I) -> Self
+    where
+        S: Into<String>,
+        I: IntoIterator<Item = (S, S)>,
+    {
+        Self {
+            runtime_config: config
+                .into_iter()
+                .map(|(k, v)| (k.into(), v.into()))
+                .collect(),
+        }
+    }
+}
+
+/// A wrapper capturing the needed internal `wasi-runtime-config` state.
+pub struct WasiRuntimeConfigView<'a> {
+    ctx: &'a WasiRuntimeConfigCtx,
+}
+
+impl<'a> WasiRuntimeConfigView<'a> {
+    /// Create a new view into the `wasi-runtime-config` state.
+    pub fn new(ctx: &'a WasiRuntimeConfigCtx) -> Self {
+        Self { ctx }
+    }
+}
```

Comment:
> > Is the idea here to separate the configured variables from the per-`Store` state, so that the variables can be reused across many `Store`s?

The idea here is very simple: it's just passing some configuration variables from the runtime to the component. I did add some documentation, but it is not described this way. Please check if the current description is reasonable. Feel free to add more comments.